### PR TITLE
Move global styles to vanilla-extract

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Move most of ui/theme/globalStyles to vanilla-extract global.css.ts

## Justify why they are needed

- Showcase of `globalStyle` API
- Compile-time styles are faster

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3904?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3894** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3894?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3901** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3901?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3879** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3879?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3904** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3904?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #3893** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3893?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3904?utm_source=stack-comment).

File: apps/store/src/pages/_app.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import Router from 'next/router'
 import { appWithTranslation } from 'next-i18next'
 import { type ReactNode } from 'react'
 import { Provider as BalancerProvider } from 'react-wrap-balancer'
+import globalCss from 'ui/src/global.css'
```
Comment (guilhermespopolin) [inline apps/store/src/pages/_app.tsx:11]:
```suggestion
import 'ui/src/global.css'
```

File: apps/store/src/pages/_app.tsx [hunk 2]
```diff
@@ -41,6 +42,11 @@ import { useForceHtmlLangAttribute } from '@/utils/l10n/useForceHtmlLangAttribut
 import { useAllowActiveStylesInSafari } from '@/utils/useAllowActiveStylesInSafari'
 import { useReloadOnCountryChange } from '@/utils/useReloadOnCountryChange'
 
+// Trick compiler into thinking we need global.css import for anything other than side effects
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const noop = (val: any) => {}
+noop(globalCss)
+
```
Comment (guilhermespopolin) [inline apps/store/src/pages/_app.tsx:49]: If you just `import ui/src/global.css` we can move away with this no?
Additionally we could remove the `export default {}` inserted into `ur/src/global.css.ts` file :)

**Review by guilhermespopolin — COMMENTED** [review]
Well done.
just left some comments.

**Review by guilhermespopolin — APPROVED** [review]
Looking good :)

**Comment by alebedev** [comment]
### Merge activity

* **Feb 8, 9:54 AM EST**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3904).

**Pull request merged**